test/suites/integration/deregister_targets_test.go

```diff
@@ -78,21 +73,32 @@ var _ = Describe("Deregister Targets", func() {
 		)
 	})
 
-	It("Kubernetes Service deletion deregisters targets", func() {
+	It("Kubernetes Service deletion triggers targets de-registering", func() {
+		Fail("Currently controller have a bug, service deletion do NOT triggers targets de-registering, need to further investigate the root cause")
 		testFramework.ExpectDeleted(ctx, service)
-		Eventually(func(g Gomega) {
-			log.Println("Verifying VPC lattice Targets deregistered")
-			targets := testFramework.GetTargets(ctx, targetGroup, deployment)
-			Expect(len(targets) == 0)
-		}).WithTimeout(5*time.Minute + 10*time.Second)
+		verifyNoTargetsForTargetGroup(targetGroup)
 	})
 
-	It("Kubernetes Deployment deletion deregisters targets", func() {
+	It("Kubernetes Deployment deletion triggers targets de-registering", func() {
 		testFramework.ExpectDeleted(ctx, deployment)
-		Eventually(func(g Gomega) {
-			log.Println("Verifying VPC lattice Targets deregistered")
-			targets := testFramework.GetTargets(ctx, targetGroup, deployment)
-			Expect(len(targets) == 0)
-		}).WithTimeout(5*time.Minute + 10*time.Second)
+		verifyNoTargetsForTargetGroup(targetGroup)
 	})
 })
+
+func verifyNoTargetsForTargetGroup(targetGroup *vpclattice.TargetGroupSummary) {
+	Eventually(func(g Gomega) {
+		log.Println("Verifying VPC lattice Targets deregistered")
+		targets, err := testFramework.LatticeClient.ListTargetsAsList(ctx, &vpclattice.ListTargetsInput{
+			TargetGroupIdentifier: targetGroup.Id,
+		})
+		g.Expect(err).To(BeNil())
+		log.Println("targets:", targets)
+		if len(targets) == 0 {
+			g.Expect(len(targets)).To(Equal(0))
+		} else {
+			for _, target := range targets {
+				g.Expect(*target.Status).To(Equal(vpclattice.TargetStatusDraining))
+			}
+		}
```

Comment:
`for _, target := range` already covers targets with len=0, you dont need if/else here